trie/smt.py

```diff
@@ -0,0 +1,266 @@
+from copy import deepcopy
+
+from eth_utils import (
+        keccak,
+        to_int,
+    )
+
+from trie.constants import (
+    BLANK_NODE,
+)
+
+from trie.exceptions import (
+        ValidationError,
+    )
+
+from trie.validation import (
+    validate_is_bytes,
+    validate_length,
+)
+
+
+def calc_root(key, value, branch):
+    path = to_int(key)
+    target_bit = 1
+    # traverse the path in leaf->root order
+    # branch is in root->leaf order (key is in MSB to LSB order)
+    node_hash = keccak(value)
+    for sibling in reversed(branch):
+        if path & target_bit:
+            node_hash = keccak(sibling + node_hash)
+        else:
+            node_hash = keccak(node_hash + sibling)
+        target_bit <<= 1
+
+    return node_hash
+
+
+class SparseMerkleProof:
+
+    def __init__(self, key, value, branch):
+        validate_is_bytes(key)
+        self._key = key
+        self._keysize = len(key)
```

Comment:
naming convention would have this be `_key_size`

Reply:
Will do